src/app/components/results-menu/scenes-list/scenes-list.component.ts

```diff
@@ -248,6 +276,32 @@ export class ScenesListComponent implements OnInit, OnDestroy {
     return Math.abs(pair[0].metadata.temporal - pair[1].metadata.temporal);
   }
 
+  public sameGranules(granules1: models.CMRProduct[], granules2: models.CMRProduct[]) {
+    const ids1 = new Set(granules1.map(granule => granule.id));
+    const ids2 = new Set(granules2.map(granule => granule.id));
+
+    return this.eqSet(ids1, ids2);
+  }
+
+  public eqSet(a1, bs) {
+    return a1.size === bs.size && this.all(this.isIn(bs), a1);
+  }
+
+  private all(pred, a1) {
+    for (const a of a1)  {
+      if (!pred(a)) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
I think array.every does what you want.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every